### PR TITLE
build: allow bbb nginx config hook for operators

### PR DESCRIPTION
**Pull request by schrd**
### What does this PR do?

Sometimes operators need to tweak the nginx config. They should not
modify config files provided by the packages. Instead they should add
their own config files in `/etc/bigbluebutton/nginx/`. This patch links
the default config provided by packages into this directory.

### Motivation

Provide a clean interface for operators to change nginx config and keep the config provided by developers for reference.

### More

nginx site configuration provided by `bbb-install-2.5.sh` needs to be changed. I'll send a PR for it.

### CODE REVIEW

**Comment by antobinary** [comment]
Nice!!
Probably would like to also delete this line: https://github.com/bigbluebutton/bigbluebutton/blob/v2.5.x-release/build/packages-template/bbb-html5/bigbluebutton.nginx#L17

**Comment by schrd** [comment]
> Nice!! Probably would like to also delete this line: https://github.com/bigbluebutton/bigbluebutton/blob/v2.5.x-release/build/packages-template/bbb-html5/bigbluebutton.nginx#L17

You're right. Sorry I missed this one.

File: build/packages-template/bbb-config/after-remove.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -e
+
+case "$1" in
+   remove|failed-upgrade|abort-upgrade|abort-install|disappear)
+   ;;
+   purge)
+      # remove file deployed by after-install script if it is still a symlink
+      if [ "x$(readlink -f /etc/bigbluebutton/nginx/include_default.nginx)" = x/usr/share/bigbluebutton/include_default.nginx ] ; then
```
Comment (danimo) [inline build/packages-template/bbb-config/after-remove.sh:8]: Not sure if we should introduce the x-prefix idiom here. It's not used anywhere else and coverse rare edge cases (a value starting in `-` being interpreted as an operator, for those who wonder). I'd rather quote the RHS.

File: build/packages-template/bbb-config/include_default.nginx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# The purpose of this file is to be included in an nginx site configuration.
+# you may copy this file to /etc/bigbluebutton/nginx/include_default.nginx and
+# adapt it to your needs
+#
+# If you are running a cluster setup, you need to change bbb-html5.nginx
+# copy it to /etc/bigbluebutting/nginx/bbb-html5.nginx and include all the
```
Comment (GuiLeme) [inline build/packages-template/bbb-config/include_default.nginx:6]: AFAIK, often times, we use clusters, wouldn't it be nicer to not have this other step? You could just do that in the after-install part, maybe? (If it doesn't break anything, of course) What do you think?

Comment (schrd) [reply build/packages-template/bbb-config/include_default.nginx:6]: this refers to https://docs.bigbluebutton.org/admin/clusterproxy.html where different location schemes have to be used. This is not related to scalelite or any other loadbalancer

**Comment by schrd** [comment]
@antobinary changing this one to 2.6 is better than changing the behaviour in a released version. I did not try 2.6 yet though.

File: build/packages-template/bbb-config/after-install.sh [hunk 3]
```diff
@@ -125,6 +125,11 @@ else
   sed -i 's/events {/worker_rlimit_nofile 10000;\n\nevents {/g' /etc/nginx/nginx.conf
 fi
 
+# symlink default bbb nginx config from package if it does not exist
+if [ ! -e /etc/bigbluebutton/nginx/include_default.nginx ] ; then
+  ln -s /usr/share/bigbluebutton/include_default.nginx /etc/bigbluebutton/nginx/include_default.nginx
```
Comment (antobinary) [inline build/packages-template/bbb-config/after-install.sh:130]: When testing locally I had to first create the directory `mkdir -p /etc/bigbluebutton/nginx`. 

```suggestion
  mkdir -p /etc/bigbluebutton/nginx
  ln -s /usr/share/bigbluebutton/include_default.nginx /etc/bigbluebutton/nginx/include_default.nginx
```

**Pull request merged**